interchaintest/helpers/clock.go

```diff
@@ -0,0 +1,120 @@
+package helpers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/strangelove-ventures/interchaintest/v7/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+// Register the clock contract
+func RegisterClockContract(t *testing.T, ctx context.Context, chain *cosmos.CosmosChain, user ibc.Wallet, contract string) {
+	cmd := []string{
+		"junod", "tx", "clock", "register", contract,
+		"--node", chain.GetRPCAddress(),
+		"--home", chain.HomeDir(),
+		"--chain-id", chain.Config().ChainID,
+		"--fees", "500ujuno",
+		"--from", user.KeyName(),
+		"--keyring-dir", chain.HomeDir(),
+		"--keyring-backend", keyring.BackendTest,
+		"-y",
+	}
+	stdout, _, err := chain.Exec(ctx, cmd, nil)
+	require.NoError(t, err)
+
+	debugOutput(t, string(stdout))
+
+	err = testutil.WaitForBlocks(ctx, 1, chain)
+	require.NoError(t, err)
+}
+
+// Unregister the clock contract
+func UnregisterClockContract(t *testing.T, ctx context.Context, chain *cosmos.CosmosChain, user ibc.Wallet, contract string) {
+	cmd := []string{
+		"junod", "tx", "clock", "unregister", contract,
+		"--node", chain.GetRPCAddress(),
+		"--home", chain.HomeDir(),
+		"--chain-id", chain.Config().ChainID,
+		"--fees", "500ujuno",
+		"--from", user.KeyName(),
+		"--keyring-dir", chain.HomeDir(),
+		"--keyring-backend", keyring.BackendTest,
+		"-y",
+	}
+	stdout, _, err := chain.Exec(ctx, cmd, nil)
+	require.NoError(t, err)
+
+	debugOutput(t, string(stdout))
+
+	err = testutil.WaitForBlocks(ctx, 1, chain)
+	require.NoError(t, err)
+}
+
+// Unjail the clock contract
+func UnjailClockContract(t *testing.T, ctx context.Context, chain *cosmos.CosmosChain, user ibc.Wallet, contract string) {
+	cmd := []string{
+		"junod", "tx", "clock", "unjail", contract,
+		"--node", chain.GetRPCAddress(),
+		"--home", chain.HomeDir(),
+		"--chain-id", chain.Config().ChainID,
+		"--fees", "500ujuno",
+		"--from", user.KeyName(),
+		"--keyring-dir", chain.HomeDir(),
+		"--keyring-backend", keyring.BackendTest,
+		"-y",
+	}
+	stdout, _, err := chain.Exec(ctx, cmd, nil)
+	require.NoError(t, err)
+
+	debugOutput(t, string(stdout))
+
+	err = testutil.WaitForBlocks(ctx, 1, chain)
+	require.NoError(t, err)
+}
+
+type ClockContract struct {
```

Comment:
note: in the future we can use the gRPC server like I do in https://github.com/strangelove-ventures/interchaintest/pull/892 for better querying with our custom modules (and automatic type handling)

Reply:
You are referring to something along the lines of this, right?

```go
// AllBalances fetches an account address's balance for all denoms it holds
func (c *CosmosChain) AllBalances(ctx context.Context, address string) (types.Coins, error) {
	params := bankTypes.QueryAllBalancesRequest{Address: address}
	grpcAddress := c.getFullNode().hostGRPCPort
	conn, err := grpc.Dial(grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, err
	}
	defer conn.Close()

	queryClient := bankTypes.NewQueryClient(conn)
	res, err := queryClient.AllBalances(ctx, &params)

	if err != nil {
		return nil, err
	}

	return res.GetBalances(), nil
}```

Reply:
yes, not required for this PR just in the future as a nice to have

(all that conn setup will not be required)

Reply:
yes, not needed for this PR but would be nice in the future (this is just a reference for you and me to remember)

Reply:
Sounds good, will keep that in mind for the future!